rllib/env/vector_env.py

```diff
@@ -248,7 +248,16 @@ def __init__(
 
     @override(VectorEnv)
     def vector_reset(self):
-        return [e.reset() for e in self.envs]
+        # Use reset_at(index) to restart and retry until
+        # we successfully create a new env.
+        resetted_obs = []
+        for i in range(len(self.envs)):
+            while True:
+                obs = self.reset_at(i)
```

Comment:
**todo**: modify reset_at() to raise exception instead of returning it instead of obs. (also document what gets returned and what error gets raised). Use `try catch` block in here:

```
while True:
     try:
         obs = self.reset_at(i)
         break
     except Exception:
         continue
```

Reply:
so this may not be ideal, but this is the best way Sven and I came up with to signal from user Env to Sampler that some Error has happened, here is the error messages, and please drop this episode from your training sample.
if you look at our Sampler code, we pipeline this all the way to CallBacks.on_episode_end().
These callbacks take either an episode (episode successful) or an Exception (episode errored out).